### PR TITLE
Adding an FAQ doc to answer common questions

### PR DESCRIPTION
**Pull request by nikhiljindal**
cc @csbell @mdelio @G-Harmon @madhusudancs

### CODE REVIEW

File: FAQs.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+We expect kubemci to be easier to use and more robust.
+
+kubemci allows users to manage multicluster ingresses without having to enroll
+all the clusters in a federation first.
+This relieves them of the overhead of managing a federation control plane
+in exchange for having to run the kubemci command explicitly each time they want
+to add or remove a cluster.
+
+Also since kubemci creates GCE resources (backend services, health checks,
+forwarding rules, etc) itself it does not have the same
```
Comment (csbell) [inline FAQs.md]: remove itself or add comma between 'itself' and 'it'

File: FAQs.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+[problem](https://github.com/kubernetes/kubernetes/issues/36327) of ingress
+controllers in each cluster fighting with each other.
```
Comment (csbell) [inline FAQs.md]: fighting -> competing with each other to program similar resources.

File: FAQs.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+specific commands for ex: get-status returning the global IP and the list of
+clusters that a multicluster ingress is spread to; or to run create/update in a
+dry-run mode first to check the changes it will make; or to run a validate
+command that validates for ex that the required services exist in all clusters
+as expected (we do not have dry-run or validate commands yet, but we can add
+them).
+
+## How do I get the self-healing properties of controller model with kubemci
+
+Users can run kubemci as a cron job so that it reads input from a data source
```
Comment (csbell) [inline FAQs.md]: I'm less fond of proposing this as a recommendation. There are multiple degrees to self-healing. Which one is this question answering? There is a lot of self-healing features built into GCLB that users may not know about and the tool doesn't preclude these.

The answer makes it sound like you should consider running kubemci in a loop for stable operation. Perhaps the better question in the FAQ is "When should I run kubemci again after kubemci create?". Then we can list the scenarios (some of which are already in your answer).

File: FAQs.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Does it work with other load balancer providers
+
+It works only with Google Cloud Load Balancer for now. In future, we can
+integrate with other load balancer providers, but to provide the same
+cross-region load balancing as GCLB, we will need to setup Geo-DNS servers that
+will route the request to the closest cluster. GCLB is able to do that using
+Anycast.
+
+## How is this different than federated ingress
```
Comment (csbell) [inline FAQs.md]: Move this to last (or third), for now.

File: FAQs.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+kubemci is a command line tool that allows users to create multicluster ingresses
+using [Google Cloud Load Balancer (GCLB)](https://cloud.google.com/load-balancing/)
+It enables users to perform [cross-region load
+balancing](https://cloud.google.com/compute/docs/load-balancing/http/cross-region-example) using
+multiple kubernetes clusters in different Google Cloud regions.
+
+## Does it work with other load balancer providers
+
+It works only with Google Cloud Load Balancer for now. In future, we can
```
Comment (csbell) [inline FAQs.md]: I think the answer needs to be two part.
1. It programs the GCLB and the many resources that are required to enable functional ingress across multiple clusters.
2. It sets up geo-awareness that is provided in GCLB through anycast. Porting it to other load balancers will require consideration of anycast or geo-dns to provide similar functionality.

File: FAQs.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+## What is kubemci?
+
+kubemci is a command line tool that allows users to create multicluster ingresses
+using [Google Cloud Load Balancer (GCLB)](https://cloud.google.com/load-balancing/)
+It enables users to perform [cross-region load
+balancing](https://cloud.google.com/compute/docs/load-balancing/http/cross-region-example) using
+multiple kubernetes clusters in different Google Cloud regions.
+
+## Does it work with other load balancer providers
```
Comment (madhusudancs) [inline FAQs.md]: "What load balancers does it work with?"

File: FAQs.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+Also since kubemci creates GCE resources (backend services, health checks,
+forwarding rules, etc) itself it does not have the same
+[problem](https://github.com/kubernetes/kubernetes/issues/36327) of ingress
+controllers in each cluster fighting with each other.
+In addition to the regular kubernetes CRUD operations, kubemci can also support
+specific commands for ex: get-status returning the global IP and the list of
+clusters that a multicluster ingress is spread to; or to run create/update in a
+dry-run mode first to check the changes it will make; or to run a validate
+command that validates for ex that the required services exist in all clusters
+as expected (we do not have dry-run or validate commands yet, but we can add
```
Comment (madhusudancs) [inline FAQs.md]: s/we can/we have plans to/

File: FAQs.md (hunk 3 above)
Comment (madhusudancs) [reply FAQs.md]: +1.

File: FAQs.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+them).
+
+## How do I get the self-healing properties of controller model with kubemci
+
+Users can run kubemci as a cron job so that it reads input from a data source
+(such as github repository) and runs kubemci commands based on that.
+Instead of a periodic job, it can also be setup as a job that watches for
+changes and runs the kubemci commands.
```
Comment (madhusudancs) [inline FAQs.md]: I think this statement should also clarify which changes users need to watch.

File: FAQs.md (hunk 3 above)
Comment (nikhiljindal) [reply FAQs.md]: Have rewritten this section. See if its any better now?

File: FAQs.md (hunk 8 above)
Comment (nikhiljindal) [reply FAQs.md]: Done

**Comment by nikhiljindal** [comment]
Thanks for the great comments @csbell and @madhusudancs.

Have pushed a new commit as per comments. PTAL.

**Comment by madhusudancs** [comment]
LGTM too.

**Comment by nikhiljindal** [comment]
Great, thanks!

**Pull request merged**